operator/v1/types_console.go

```diff
@@ -194,7 +194,7 @@ type DeveloperConsoleCatalogCustomization struct {
 	// types allows enabling or disabling of sub-catalog types that user can see in the Developer catalog.
 	// When omitted, all the sub-catalog types will be shown.
 	// +optional
-	Types DeveloperConsoleCatalogTypesState `json:"types"`
+	Types *DeveloperConsoleCatalogTypes `json:"types,omitempty"`
```

Comment:
Did you try with just omitempty and without the pointer? We have used this successfully in other areas of the API codebase. The pointer just adds complexity

Reply:
@JoelSpeed Yes we tried first with `omitempty` only but had the same e2e CI issues. After that we added the pointer and now its working fine.